app/src/main/java/io/pslab/activity/PowerSourceActivity.java

```diff
@@ -516,23 +517,94 @@ protected void onSaveInstanceState(@NonNull Bundle outState) {
-            public void onStartTrackingTouch(Croller croller) {/**/}
+                @Override
+                public void onStartTrackingTouch(Croller croller) { /**/ }
+
+                @Override
+                public void onStopTrackingTouch(Croller croller) {
+                    setPower(pin);
+                    switch (pin) {
+                        case PV1:
+                            if ((int) (progress * 3.3 / 10) > PV3_CONTROLLER_MAX) {
+                                controllerPV3.setProgress(PV3_CONTROLLER_MAX);
+                            }
+                            else if ((int) (progress * 3.3 / 10) < CONTROLLER_MIN) {
+                                controllerPV3.setProgress(CONTROLLER_MIN);
+                            }
+                            else {
+                                controllerPV3.setProgress((int) (progress * 3.3 / 10));
+                            }
+                            break;
+                        case PV2:
+                            if (PCS_CONTROLLER_MAX - (progress / 2) > PCS_CONTROLLER_MAX) {
+                                controllerPCS.setProgress(PCS_CONTROLLER_MAX);
+                            }
+                            else if (PCS_CONTROLLER_MAX - (progress / 2) < CONTROLLER_MIN) {
+                                controllerPCS.setProgress(CONTROLLER_MIN);
+                            }
+                            else {
+                                controllerPCS.setProgress(PCS_CONTROLLER_MAX - (progress / 2));
+                            }
+                            break;
+                        case PV3:
+                            if ((int) (progress * 10 / 3.3) > PV1_CONTROLLER_MAX) {
+                                controllerPV1.setProgress(PV1_CONTROLLER_MAX);
+                            }
+                            else if ((int) (progress * 10 / 3.3) < CONTROLLER_MIN) {
+                                controllerPV1.setProgress(CONTROLLER_MIN);
+                            }
+                            else {
+                                controllerPV1.setProgress((int) (progress * 10 / 3.3));
+                            }
+                            break;
+                        case PCS:
+                            if (PV2_CONTROLLER_MAX - (progress * 2) > PV2_CONTROLLER_MAX) {
+                                controllerPV2.setProgress(PV2_CONTROLLER_MAX);
+                            }
+                            else if (PV2_CONTROLLER_MAX - (progress * 2) < CONTROLLER_MIN) {
+                                controllerPV2.setProgress(CONTROLLER_MIN);
+                            }
+                            else {
+                                controllerPV2.setProgress(PV2_CONTROLLER_MAX - (progress * 2));
+                            }
+                            break;
+                        default:
+                            break;
+                    }
+                }
+            });
+        } else {
```

Comment:
Open question: How about replacing this `else` with an `if (CommunicationHandler.PSLAB_VERSION == 6)`?

Then we could add another `else` after this block to catch all unknown versions (e.g. a future version 7) instead of using the PsLab v5 way to connect to them which will probably fail.